File: protoqueue-compiler/src/main/java/net/port/transformer/compiler/common/CompilerContext.java
package net.port.transformer.compiler.common;

import javax.annotation.processing.ProcessingEnvironment;

/**
 * Created by zhongyongsheng on 2018/4/13.
 */
public class CompilerContext {

    public static CompilerContext defaultIntance;
    public ProcessingEnvironment processingEnvironment;
    public Log log;

    public CompilerContext(ProcessingEnvironment env) {
        processingEnvironment = env;
        log = new Log(env.getMessager(), null);
    }
}

File: protoqueue-compiler/src/main/java/net/protoqueue/compiler/common/CompilerContext.kt
package net.protoqueue.compiler.common

import javax.annotation.processing.ProcessingEnvironment

/**
 * 保存编译的上下文信息
 * Created by zhongyongsheng on 2018/4/13.
 */
object CompilerContext {
    lateinit var log: Log
    lateinit var processingEnvironment: ProcessingEnvironment

    fun init(processingEnvironment: ProcessingEnvironment) {
        this.processingEnvironment = processingEnvironment
        this.log = Log(processingEnvironment.messager)
    }
}
